content/departments/cloud/technical-docs/scalability_testing.md

```diff
@@ -0,0 +1,98 @@
+## Running scalability tests on Cloud instances
+
+This document lists steps required to verify that a Cloud instance (_target instance_) can pass [Sourcegraph scalability tests](https://github.com/sourcegraph/k6), with the following assumptions:
+
+- we are testing scale of [Size L](https://docs.google.com/spreadsheets/d/1n-KfGc8m1w09rIzNKm5tRxAYmP4-w11CVOCplMvVazk/edit#gid=1172385107&range=B7) (previously Size S) customers -> 10000 users / 100000 repos
+- testing is performed using an existing Cloud instance (`rctest` in links below)
+- load generator uses external Cloud endpoints (so requests go through Cloudflare and GCLB)
+
+### Preparation
+
+1. Make sure target instance is of size `n2-standard-32` (or, has at least 32 CPU)
+2. Make sure target instance has a SSD data disk with capacity >=1000GB
+3. Create a Code Host connection with >=100000 repos - [example for Github](https://github.com/sourcegraph/reference-architecture-test/blob/main/configs/github_config_100k_repos.json)
+4. Wait until your instance has [cloned](https://rctest.sourcegraph.com/-/debug/grafana/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Prometheus%22,%7B%22exemplar%22:true,%22expr%22:%22src_gitserver_repo_count%22,%22requestId%22:%22Q-662fcf77-5726-4d55-90cf-e523b29d1ea3-0A%22%7D%5D) and [indexed]([index_num_indexed](https://rctest.sourcegraph.com/-/debug/grafana/explore?orgId=1&left=%5B%22now-7d%22,%22now%22,%22Prometheus%22,%7B%22exemplar%22:true,%22expr%22:%22index_num_indexed%22,%22requestId%22:%22Q-bab9d4c3-db12-4ae7-b051-2be6ec528e58-0A%22%7D%5D) all repos.
+5. To generate load, create a 4 CPU VM in the same region as your target instance
```

Comment:
Why in the same region if we go through external LB?

Reply:
To avoid cross-region latency affecting test results (we expect a certain latency distribution on the client, and it will be different if ex. testing eu from us).